driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -242,29 +242,30 @@ OperationResult executeListCollections(final BsonDocument operation) {
         ListCollectionsIterable<BsonDocument> iterable = session == null
                 ? database.listCollections(BsonDocument.class)
                 : database.listCollections(session, BsonDocument.class);
-        for (Map.Entry<String, BsonValue> cur : arguments.entrySet()) {
-            switch (cur.getKey()) {
-                case "session":
-                    break;
-                case "filter":
-                    iterable.filter(cur.getValue().asDocument());
-                    break;
-                case "batchSize":
-                    iterable.batchSize(cur.getValue().asNumber().intValue());
-                    break;
-                case "timeoutMode":
-                    setTimeoutMode(iterable, cur);
-                    break;
-                case "maxTimeMS":
-                    iterable.maxTime(cur.getValue().asNumber().longValue(), TimeUnit.MILLISECONDS);
-                    break;
-                default:
-                    throw new UnsupportedOperationException("Unsupported argument: " + cur.getKey());
+        return resultOf(() -> {
+            for (Map.Entry<String, BsonValue> cur : arguments.entrySet()) {
+                switch (cur.getKey()) {
+                    case "session":
+                        break;
+                    case "filter":
+                        iterable.filter(cur.getValue().asDocument());
+                        break;
+                    case "batchSize":
+                        iterable.batchSize(cur.getValue().asNumber().intValue());
+                        break;
+                    case "timeoutMode":
+                        setTimeoutMode(iterable, cur);
```

Comment:
Setting the timeout mode can now throw an exception - so running in the resultOf block.